File: Week10/SoyutSınıfJava/src/pkt/Daire.java
package pkt;

public class Daire extends Sekil {
	private double yaricap;
	
	public Daire(double yaricap,String renk) {
		super(renk);
		this.yaricap = yaricap;
	}
	
	@Override
	public double alan() {
		return Math.PI*Math.pow(yaricap, 2);
	}
	@Override
	public double cevre() {
		return 2*Math.PI*yaricap;
	}
}

File: Week10/SoyutBenzetme/src/Daire.c
#include "Daire.h"

Daire DaireOlustur(double yaricap,char* renk){
	Daire this;
	this = (Daire)malloc(sizeof(struct DAIRE));
	this->super = SekilOlustur(renk);
	this->yaricap = yaricap;
	this->super->alan = &Alan;
	this->super->cevre = &Cevre;
	this->yoket = &DaireYoket;
	return this;
}
double Alan(const Daire this){
	return M_PI*pow(this->yaricap,2);
}
double Cevre(const Daire this){
	return 2*M_PI*this->yaricap;
}
void DaireYoket(const Daire this){
	if(this == NULL) return;
	this->super->yoket(this->super);
	free(this);
}
